stdcommands/forex/forex.go

```diff
@@ -36,112 +38,108 @@ var Command = &commands.YAGCommand{
 
 	RunFunc: func(data *dcmd.Data) (interface{}, error) {
 		amount := data.Args[0].Float64()
-		check, err := requestAPI("https://api.exchangerate.host/symbols")
+		var currenciesResult Currencies
+		var exchangeRateResult ExchangeRate
+
+		err := requestAPI(currenciesAPIURL, &currenciesResult)
 		if err != nil {
 			return nil, err
 		}
-		from := check.Symbols[strings.ToUpper(data.Args[1].Str())]
-		to := check.Symbols[strings.ToUpper(data.Args[2].Str())]
-		// Checks the max amount of pages by the number of symbols on each page (15)
-		maxPages := int(math.Ceil(float64(len(check.Symbols)) / float64(15)))
-		if (to == nil) || (from == nil) {
+
+		from := strings.ToUpper(data.Args[1].Str())
+		to := strings.ToUpper(data.Args[2].Str())
+
+		// Check if the currencies exist in the map
+		_, fromExist := currenciesResult[from]
+		_, toExist := currenciesResult[to]
+
+		// Checks the max amount of pages by the number of symbols on each page
+		maxPages := int(math.Ceil(float64(len(currenciesResult)) / float64(currencyPerPage)))
+
+		// If the currency isn't supported by API.
+		if !toExist || !fromExist {
 			_, err = paginatedmessages.CreatePaginatedMessage(
 				data.GuildData.GS.ID, data.ChannelID, 1, maxPages, func(p *paginatedmessages.PaginatedMessage, page int) (*discordgo.MessageEmbed, error) {
-					return errEmbed(check, page)
+					embed, err := errEmbed(currenciesResult, page)
+					if err != nil {
+						return nil, err
+					}
+					return embed, nil
 				})
 			if err != nil {
 				return nil, err
 			}
 			return nil, nil
 		}
-		output, err := requestAPI(fmt.Sprintf("https://api.exchangerate.host/convert?from=%s&to=%s&amount=%.3f", from.Code, to.Code, amount))
+
+		err = requestAPI(fmt.Sprintf("https://api.frankfurter.app/latest?amount=%.3f&from=%s&to=%s", amount, from, to), &exchangeRateResult)
 		if err != nil {
 			return nil, err
 		}
+
 		p := message.NewPrinter(language.English)
 		embed := &discordgo.MessageEmbed{
 			Title:       "💱Currency Exchange Rate",
-			Description: p.Sprintf("\n%.2f **%s** (%s) is %.3f **%s** (%s).", amount, from.Description, output.Query.From, output.Result, to.Description, output.Query.To),
+			Description: p.Sprintf("\n%.2f **%s** (%s) is %.3f **%s** (%s).", amount, currenciesResult[from], from, exchangeRateResult.Rates[to], currenciesResult[to], to),
 			Color:       0xAE27FF,
-			Footer:      &discordgo.MessageEmbedFooter{Text: fmt.Sprintf("Based on currency rate 1 : %f", output.Info.Rate)},
 			Timestamp:   time.Now().UTC().Format(time.RFC3339),
 		}
 		return embed, nil
 	},
 }
 
-func requestAPI(query string) (*ExchangeAPIResult, error) {
+func requestAPI(query string, result interface{}) error {
 	req, err := http.NewRequest("GET", query, nil)
```

Comment:
While we're at nitpicking some (minor) details:
```suggestion
	req, err := http.NewRequest(http.MethodGet, query, nil)
```